# California State-Level Funding for Schools

This is a general overview of how public schools in California recieve government funding for education, which is divided into the specific educational programs. The main source of funding for a school is known as the Principal Apportionment.

#### The __Principal Apportionment__ provides categorical funding for LEAs, including... 
1. Local Control Funding Formula (LCFF) -- this is the primary source of general purpose funding for an LEA 
2. Special Education Programs
3. Expanded Learning Opportunities Program (Grades TK - 6th)
4. Arts and Music for Schools
5. Equity Multiplier

How much funding a school gets is determined by various factors, including the equity multiplier and the amount of children who actually attend the school everyday, a metric called ADA, or Average Daily Attendance.

#### __Key Terms__:
* LEA: _Local Educational Agency_ 
    - an LEA is the public authority responsible for providing education in an area; typically a school, county office of education, or public school district.
* ADA: _Average Daily Attendance_
    - This is a measurement of how many kids on average attend a school--this goes into the formula which determines how much funding a school gets.
* AB: _[California] Assembly Bill_
    - Specific bills determine how funding is sourced, categorized and allocated.
* SELPA: _Special Education Local Plan Area_
    - Coordinators of Special Education for a certain area, sometimes a district.
* LCFF: _Local Control Funding Formula_
    - Main funding group for CA schools; covers teacher salaries, equipment, etc. 

#### __Data Sources__:
 1. _California Department of Education: Principal Apportionments for the 2024-2025 School Year_
    - https://www.cde.ca.gov/fg/aa/pa/iassf24p1.asp 
 2. _California Special Education Local Plan Areas_
    - https://www.cde.ca.gov/sp/se/as/caselpas.asp
    
#### __Informational Sources__:
 1. _California Assembly Bill 602 - Special Education_
    - https://www.cde.ca.gov/fg/aa/se/ab602apptdat.asp 


# Data Analysis

The overall goal of the analysis is to visualize how much government and grant funding public schools and districts recieve in California, and how much of these funds are allocated to the schools' various special education programs.

In [109]:
import pandas as pd
import matplotlib.pyplot as plt

In [83]:
# this dataframe includes the Principal Apportionments (PAs) for the 2024-25 academic year.
pa24 = pd.read_csv('pa-summary-24p1.csv', encoding='latin1')

In [84]:
# data cleaning
pa24.columns = pa24.iloc[3]
# first & last few rows are descriptions
pa24 = pa24.iloc[4:-4]
pa24.reset_index(drop=True, inplace=True)
print(pa24.columns)

Index(['County Code', 'District Code', 'School Code', 'Charter Number',
       'Charter Fund Type', 'Local Educational Agency', 'District Type',
       'County Office of Education LCFF State Aid \n(0000-8011)\nA-1',
       'School District LCFF State Aid\n (0000-8011)\nA-2',
       'Charter School LCFF State Aid\n (0000-8011)\nA-3',
       'School District Basic Aid "Choice"\n (0000-8011)\nA-4',
       'School District Basic Aid Court-Ordered Voluntary Pupil Transfer \n(0000-8011)\nA-5',
       'School District Basic Aid Open Enrollment \n(0000-8011)\nA-6',
       'School District Basic Aid Supplement Charter School Adjustment \n(0000-8011)\nA-7',
       'Adults in Correctional Facilities \n(6015-8311)\nA-8',
       'Special Education \nAB 602\n(6500-8311)\nA-9',
       'Special Education\nInfants 0-2 \n(6510-8311)\nA-10',
       'Special Education\nMental Health Services\n(6546-8590)\nA-11',
       'Special Education\nEarly Intervention Preschool Grant\n(6547-8590)\nA-12',
       'Exp

### High School Funding Data Exploration

In [85]:
# just looking at High Schools 
hs = pa24.where(pa24['District Type'] == 'HIGH').dropna()

In [86]:
# looking at how much special education funding High School Districts get from their first Principal Apportionment
cols = ['Local Educational Agency', 'District Type',
       "Special Education \nAB 602\n(6500-8311)\nA-9"]
hs[cols].where(hs["Special Education \nAB 602\n(6500-8311)\nA-9"].str.strip() != "-").dropna()

3                              Local Educational Agency District Type  \
70                                   Latitude 37.8 High          HIGH   
180                               Pacific Crest Academy          HIGH   
268   Northcoast Preparatory and Performing Arts Aca...          HIGH   
277                             Six Rivers Charter High          HIGH   
278                                 Laurel Tree Charter          HIGH   
...                                                 ...           ...   
2177                           Accelerated Charter High          HIGH   
2203                       Gold Rush Home Study Charter          HIGH   
2204     Connections Visual and Performing Arts Academy          HIGH   
2224         Camarillo Academy of Progressive Education          HIGH   
2225  Architecture, Construction & Engineering Chart...          HIGH   

3    Special Education \nAB 602\n(6500-8311)\nA-9  
70                                              0  
180                                             0  
268                                             0  
277                                             0  
278                                             0  
...                                           ...  
2177                                            0  
2203                                            0  
2204                                            0  
2224                                            0  
2225                                            0  

[72 rows x 3 columns]

#### Discussion

The _Special Education: AB 602_ column appears to be completey empty in the hs dataframe because not all High School Districts are a SELPA, or Special Education Local Plan Area. 

SELPAs are the coordinators who ensure all students in CA recieve free special education services within their LEA. 

Finding the entries which contain a value for this column will be useful in understanding how AB 602 funds are distributed.

In [94]:
ab602 = "Special Education \nAB 602\n(6500-8311)\nA-9"

selpa_funding = pa24[pa24[ab602] != '0']

selpa_funding

3    County Code District Code School Code Charter Number Charter Fund Type  \
18            01         61127     0000000            NaN               NaN   
20            01         61150     0000000            NaN               NaN   
22            01         61176     0000000            NaN               NaN   
33            01         61259     0000000            NaN               NaN   
68            01         75101     0000000            NaN               NaN   
...          ...           ...         ...            ...               ...   
2193          55         10553     0000000            NaN               NaN   
2207          56         10561     0000000            NaN               NaN   
2239          57         10579     0000000            NaN               NaN   
2252          58         10587     0000000            NaN               NaN   
2264       TOTAL           NaN         NaN            NaN               NaN   

3                      Local Educational Agency District Type  \
18                          Albany City Unified       UNIFIED   
20                        Castro Valley Unified       UNIFIED   
22                              Fremont Unified       UNIFIED   
33                              Oakland Unified       UNIFIED   
68                           Pleasanton Unified       UNIFIED   
...                                         ...           ...   
2193  Tuolumne County Superintendent of Schools     CO OFFICE   
2207            Ventura Co. Office of Education     CO OFFICE   
2239               Yolo Co. Office of Education     CO OFFICE   
2252               Yuba Co. Office of Education     CO OFFICE   
2264                                        NaN           NaN   

3    County Office of Education LCFF State Aid \n(0000-8011)\nA-1  \
18                                                    0             
20                                                    0             
22                                                    0             
33                                                    0             
68                                                    0             
...                                                 ...             
2193                                            6148837             
2207                                           16571751             
2239                                            5613684             
2252                                            4396588             
2264                                          740000620             

3    School District LCFF State Aid\n (0000-8011)\nA-2  \
18                                            20003593   
20                                            55966240   
22                                           159356595   
33                                           256500027   
68                                            58442631   
...                                                ...   
2193                                                 0   
2207                                                 0   
2239                                                 0   
2252                                                 0   
2264                                       38246131573   

3    Charter School LCFF State Aid\n (0000-8011)\nA-3  ...  \
18                                                  0  ...   
20                                                  0  ...   
22                                                  0  ...   
33                                                  0  ...   
68                                                  0  ...   
...                                               ...  ...   
2193                                                0  ...   
2207                                                0  ...   
2239                                                0  ...   
2252                                                0  ...   
2264                                       6319315906  ...   


### Data Validation

There are 128 rows in this dataframe (including a 'total' row), indicating there are 127 SELPAs in California; a quick search should be performed to confirm that is accurate.

The CDE website also has a list of SELPAs which can be cross-referenced, found here: https://www.cde.ca.gov/sp/se/as/caselpas.asp 

In [95]:
selpas = pd.read_csv('SELPAs.csv', encoding='latin1')

In [96]:
print(len(selpas))

136


### SELPAs

There are 136 SELPAs, meaning the pa24 dataframe could have 9 missing values. Since the numbers are very close, the funding information should be accurate.

In [102]:
# total special education funding 
selpa_funding[ab602].iloc[-1]

'4506685138'

In [110]:
# the max amount in the 'TOTAL' row is the total PA (Principal Apportionment)
max(pa24.iloc[-1, 7:-2].astype(int))

56135473411

## Isolating Group Totals for Visualization

In [147]:
# totals for visualization
pd.set_option('display.max_colwidth', None) 

totals = pa24.iloc[-1, 7:-2].reset_index()
totals.columns = ['Funding Area', 'Amount in $']
totals['Amount in $'] = totals['Amount in $'].astype(int)
totals

Funding Area  \
0                                                                     County Office of Education LCFF State Aid \n(0000-8011)\nA-1   
1                                                                                School District LCFF State Aid\n (0000-8011)\nA-2   
2                                                                                 Charter School LCFF State Aid\n (0000-8011)\nA-3   
3                                                                            School District Basic Aid "Choice"\n (0000-8011)\nA-4   
4                                              School District Basic Aid Court-Ordered Voluntary Pupil Transfer \n(0000-8011)\nA-5   
5                                                                     School District Basic Aid Open Enrollment \n(0000-8011)\nA-6   
6                                                School District Basic Aid Supplement Charter School Adjustment \n(0000-8011)\nA-7   
7                                                                             Adults in Correctional Facilities \n(6015-8311)\nA-8   
8                                                                                     Special Education \nAB 602\n(6500-8311)\nA-9   
9                                                                               Special Education\nInfants 0-2 \n(6510-8311)\nA-10   
10                                                                    Special Education\nMental Health Services\n(6546-8590)\nA-11   
11                                                        Special Education\nEarly Intervention Preschool Grant\n(6547-8590)\nA-12   
12                                                                      Expanded Learning Opportunities Program\n(2600-8590)\nA-13   
13                 Transfer of Funds for County Served District Funded ADA\n(0000-7142 for districts and 0000-8710 for COEs)\nA-14   
14                                                         Other State Aid (Ch.44/2021, Ch.48/2023, and 38/2024) (7810-8590)\nA-15   
15                                                                  Home-to-School Transportation Reimbursement\n(0000-8590)\nA-16   
16                                                                     Proposition 28: Arts and Music In Schools (6770-8590)\nA-17   
17                                                                                        LCFF Equity Multiplier (7399-8590)\nA-18   
18                                                                   Student Support and Enrichment Block Grant (6018-8590) \nA-19   
19                                                                   New or Expanding Charter Advance Payments \n(0000-8011)\nA-20   
20  School District Advance Payments of In-lieu of Property Tax Transfers for New or Expanding Charter Schools \n(0000-8011)\nA-21   
21                                                                                                   Amount Charter Overpaid\nA-22   
22                                                                                                PY Amount Charter Overpaid\nA-23   
23                                                                                         Adjustments and PY Recomputations\nA-24   
24                                                                   Total Principal Apportionment\n(Sum of A-1 through A-24)\nB-1   
25                                                                                               EPA Entitlement\n(1400-8012)\nC-1   
26                                                       Total of LCFF State Aid and EPA Entitlement\n(A-1 + A-2 + A-3 + C-1)\nD-1   

    Amount in $  
0     740000620  
1   38246131573  
2    6319315906  
3       1394567  
4       6241697  
5             0  
6      50265280  
7       2983118  
8    4506685138  
9      90551361  
10    453169000  
11    259999998  
12   3995000000  
13            0  
14      8600000  
15    720630821  
16    907030402  
17    303210000  
18     31242360  
19   

In [148]:
charter_rows = ["Charter School LCFF State Aid\n (0000-8011)\nA-3", 
                "New or Expanding Charter Advance Payments \n(0000-8011)\nA-20", 
                "School District Advance Payments of In-lieu of Property Tax Transfers for New or Expanding Charter Schools \n(0000-8011)\nA-21",
                "Amount Charter Overpaid\nA-22", "PY Amount Charter Overpaid\nA-23"]

charter_filtered = (totals[totals["Funding Area"].isin(charter_rows)])

new_row = {"Funding Area" : "Charter School LCFF State Aid - Adjustments\n (0000-8011)\nA-3",
           "Amount in $" : sum(charter_filtered['Amount in $'])}

totals = pd.concat([totals, pd.DataFrame([new_row])], ignore_index=True)

In [149]:
totals

Funding Area  \
0                                                                     County Office of Education LCFF State Aid \n(0000-8011)\nA-1   
1                                                                                School District LCFF State Aid\n (0000-8011)\nA-2   
2                                                                                 Charter School LCFF State Aid\n (0000-8011)\nA-3   
3                                                                            School District Basic Aid "Choice"\n (0000-8011)\nA-4   
4                                              School District Basic Aid Court-Ordered Voluntary Pupil Transfer \n(0000-8011)\nA-5   
5                                                                     School District Basic Aid Open Enrollment \n(0000-8011)\nA-6   
6                                                School District Basic Aid Supplement Charter School Adjustment \n(0000-8011)\nA-7   
7                                                                             Adults in Correctional Facilities \n(6015-8311)\nA-8   
8                                                                                     Special Education \nAB 602\n(6500-8311)\nA-9   
9                                                                               Special Education\nInfants 0-2 \n(6510-8311)\nA-10   
10                                                                    Special Education\nMental Health Services\n(6546-8590)\nA-11   
11                                                        Special Education\nEarly Intervention Preschool Grant\n(6547-8590)\nA-12   
12                                                                      Expanded Learning Opportunities Program\n(2600-8590)\nA-13   
13                 Transfer of Funds for County Served District Funded ADA\n(0000-7142 for districts and 0000-8710 for COEs)\nA-14   
14                                                         Other State Aid (Ch.44/2021, Ch.48/2023, and 38/2024) (7810-8590)\nA-15   
15                                                                  Home-to-School Transportation Reimbursement\n(0000-8590)\nA-16   
16                                                                     Proposition 28: Arts and Music In Schools (6770-8590)\nA-17   
17                                                                                        LCFF Equity Multiplier (7399-8590)\nA-18   
18                                                                   Student Support and Enrichment Block Grant (6018-8590) \nA-19   
19                                                                   New or Expanding Charter Advance Payments \n(0000-8011)\nA-20   
20  School District Advance Payments of In-lieu of Property Tax Transfers for New or Expanding Charter Schools \n(0000-8011)\nA-21   
21                                                                                                   Amount Charter Overpaid\nA-22   
22                                                                                                PY Amount Charter Overpaid\nA-23   
23                                                                                         Adjustments and PY Recomputations\nA-24   
24                                                                   Total Principal Apportionment\n(Sum of A-1 through A-24)\nB-1   
25                                                                                               EPA Entitlement\n(1400-8012)\nC-1   
26                                                       Total of LCFF State Aid and EPA Entitlement\n(A-1 + A-2 + A-3 + C-1)\nD-1   
27                                                                  Charter School LCFF State Aid - Adjustments\n (0000-8011)\nA-3   

    Amount in $  
0     740000620  
1   38246131573  
2    6319315906  
3       1394567  
4       6241697  
5             0  
6      50265280  
7       2983118  
8    4506685138  
9      90551361  
10    453169000  
11    259999998

In [150]:
# drop summed rows
totals = (totals[~totals["Funding Area"].isin(charter_rows)]).reset_index()
totals = totals.drop(columns=["index"])

In [151]:
totals

Funding Area  \
0                                                      County Office of Education LCFF State Aid \n(0000-8011)\nA-1   
1                                                                 School District LCFF State Aid\n (0000-8011)\nA-2   
2                                                             School District Basic Aid "Choice"\n (0000-8011)\nA-4   
3                               School District Basic Aid Court-Ordered Voluntary Pupil Transfer \n(0000-8011)\nA-5   
4                                                      School District Basic Aid Open Enrollment \n(0000-8011)\nA-6   
5                                 School District Basic Aid Supplement Charter School Adjustment \n(0000-8011)\nA-7   
6                                                              Adults in Correctional Facilities \n(6015-8311)\nA-8   
7                                                                      Special Education \nAB 602\n(6500-8311)\nA-9   
8                                                                Special Education\nInfants 0-2 \n(6510-8311)\nA-10   
9                                                      Special Education\nMental Health Services\n(6546-8590)\nA-11   
10                                         Special Education\nEarly Intervention Preschool Grant\n(6547-8590)\nA-12   
11                                                       Expanded Learning Opportunities Program\n(2600-8590)\nA-13   
12  Transfer of Funds for County Served District Funded ADA\n(0000-7142 for districts and 0000-8710 for COEs)\nA-14   
13                                          Other State Aid (Ch.44/2021, Ch.48/2023, and 38/2024) (7810-8590)\nA-15   
14                                                   Home-to-School Transportation Reimbursement\n(0000-8590)\nA-16   
15                                                      Proposition 28: Arts and Music In Schools (6770-8590)\nA-17   
16                                                                         LCFF Equity Multiplier (7399-8590)\nA-18   
17                                                    Student Support and Enrichment Block Grant (6018-8590) \nA-19   
18                                                                          Adjustments and PY Recomputations\nA-24   
19                                                    Total Principal Apportionment\n(Sum of A-1 through A-24)\nB-1   
20                                                                                EPA Entitlement\n(1400-8012)\nC-1   
21                                        Total of LCFF State Aid and EPA Entitlement\n(A-1 + A-2 + A-3 + C-1)\nD-1   
22                                                   Charter School LCFF State Aid - Adjustments\n (0000-8011)\nA-3   

    Amount in $  
0     740000620  
1   38246131573  
2       1394567  
3       6241697  
4             0  
5      50265280  
6       2983118  
7    4506685138  
8      90551361  
9     453169000  
10    259999998  
11   3995000000  
12            0  
13      8600000  
14    720630821  
15    907030402  
16    303210000  
17     31242360  
18   -479713692  
19  56135473411  
20   8522429192  
21  53827877291  
22   6292051168

In [165]:
# drop negative row & totals
totals = totals.drop(totals.index[18:22])
totals.reset_index()

index  \
0       0   
1       1   
2       2   
3       3   
4       4   
5       5   
6       6   
7       7   
8       8   
9       9   
10     10   
11     11   
12     12   
13     13   
14     14   
15     15   
16     16   
17     17   

                                                                                                       Funding Area  \
0                                                      County Office of Education LCFF State Aid \n(0000-8011)\nA-1   
1                                                                 School District LCFF State Aid\n (0000-8011)\nA-2   
2                                                             School District Basic Aid "Choice"\n (0000-8011)\nA-4   
3                               School District Basic Aid Court-Ordered Voluntary Pupil Transfer \n(0000-8011)\nA-5   
4                                                      School District Basic Aid Open Enrollment \n(0000-8011)\nA-6   
5                                 School District Basic Aid Supplement Charter School Adjustment \n(0000-8011)\nA-7   
6                                                              Adults in Correctional Facilities \n(6015-8311)\nA-8   
7                                                                      Special Education \nAB 602\n(6500-8311)\nA-9   
8                                                                Special Education\nInfants 0-2 \n(6510-8311)\nA-10   
9                                                      Special Education\nMental Health Services\n(6546-8590)\nA-11   
10                                         Special Education\nEarly Intervention Preschool Grant\n(6547-8590)\nA-12   
11                                                       Expanded Learning Opportunities Program\n(2600-8590)\nA-13   
12  Transfer of Funds for County Served District Funded ADA\n(0000-7142 for districts and 0000-8710 for COEs)\nA-14   
13                                          Other State Aid (Ch.44/2021, Ch.48/2023, and 38/2024) (7810-8590)\nA-15   
14                                                   Home-to-School Transportation Reimbursement\n(0000-8590)\nA-16   
15                                                      Proposition 28: Arts and Music In Schools (6770-8590)\nA-17   
16                                                                         LCFF Equity Multiplier (7399-8590)\nA-18   
17                                                    Student Support and Enrichment Block Grant (6018-8590) \nA-19   

    Amount in $  
0     740000620  
1   38246131573  
2       1394567  
3       6241697  
4             0  
5      50265280  
6       2983118  
7    4506685138  
8      90551361  
9     453169000  
10    259999998  
11   3995000000  
12            0  
13      8600000  
14    720630821  
15    907030402  
16    303210000  
17     31242360

In [166]:
totals

Funding Area  \
0                                                      County Office of Education LCFF State Aid \n(0000-8011)\nA-1   
1                                                                 School District LCFF State Aid\n (0000-8011)\nA-2   
2                                                             School District Basic Aid "Choice"\n (0000-8011)\nA-4   
3                               School District Basic Aid Court-Ordered Voluntary Pupil Transfer \n(0000-8011)\nA-5   
4                                                      School District Basic Aid Open Enrollment \n(0000-8011)\nA-6   
5                                 School District Basic Aid Supplement Charter School Adjustment \n(0000-8011)\nA-7   
6                                                              Adults in Correctional Facilities \n(6015-8311)\nA-8   
7                                                                      Special Education \nAB 602\n(6500-8311)\nA-9   
8                                                                Special Education\nInfants 0-2 \n(6510-8311)\nA-10   
9                                                      Special Education\nMental Health Services\n(6546-8590)\nA-11   
10                                         Special Education\nEarly Intervention Preschool Grant\n(6547-8590)\nA-12   
11                                                       Expanded Learning Opportunities Program\n(2600-8590)\nA-13   
12  Transfer of Funds for County Served District Funded ADA\n(0000-7142 for districts and 0000-8710 for COEs)\nA-14   
13                                          Other State Aid (Ch.44/2021, Ch.48/2023, and 38/2024) (7810-8590)\nA-15   
14                                                   Home-to-School Transportation Reimbursement\n(0000-8590)\nA-16   
15                                                      Proposition 28: Arts and Music In Schools (6770-8590)\nA-17   
16                                                                         LCFF Equity Multiplier (7399-8590)\nA-18   
17                                                    Student Support and Enrichment Block Grant (6018-8590) \nA-19   

    Amount in $  
0     740000620  
1   38246131573  
2       1394567  
3       6241697  
4             0  
5      50265280  
6       2983118  
7    4506685138  
8      90551361  
9     453169000  
10    259999998  
11   3995000000  
12            0  
13      8600000  
14    720630821  
15    907030402  
16    303210000  
17     31242360

In [167]:
# export table for Tableau Visualization
totals.to_csv('CA-Funding-Group-Totals.csv', index=False)

### Categories for Summary Table
Isolating totals of some of the main funding categories for the Tableau visualization 

In [170]:
# summary of totals for viz
pd.set_option('display.max_colwidth', None) 

summary = pa24.iloc[-1, 7:-2].reset_index()
summary.columns = ['Funding Area', 'Amount in $']
summary['Amount in $'] = summary['Amount in $'].astype(int)
summary

Funding Area  \
0                                                                     County Office of Education LCFF State Aid \n(0000-8011)\nA-1   
1                                                                                School District LCFF State Aid\n (0000-8011)\nA-2   
2                                                                                 Charter School LCFF State Aid\n (0000-8011)\nA-3   
3                                                                            School District Basic Aid "Choice"\n (0000-8011)\nA-4   
4                                              School District Basic Aid Court-Ordered Voluntary Pupil Transfer \n(0000-8011)\nA-5   
5                                                                     School District Basic Aid Open Enrollment \n(0000-8011)\nA-6   
6                                                School District Basic Aid Supplement Charter School Adjustment \n(0000-8011)\nA-7   
7                                                                             Adults in Correctional Facilities \n(6015-8311)\nA-8   
8                                                                                     Special Education \nAB 602\n(6500-8311)\nA-9   
9                                                                               Special Education\nInfants 0-2 \n(6510-8311)\nA-10   
10                                                                    Special Education\nMental Health Services\n(6546-8590)\nA-11   
11                                                        Special Education\nEarly Intervention Preschool Grant\n(6547-8590)\nA-12   
12                                                                      Expanded Learning Opportunities Program\n(2600-8590)\nA-13   
13                 Transfer of Funds for County Served District Funded ADA\n(0000-7142 for districts and 0000-8710 for COEs)\nA-14   
14                                                         Other State Aid (Ch.44/2021, Ch.48/2023, and 38/2024) (7810-8590)\nA-15   
15                                                                  Home-to-School Transportation Reimbursement\n(0000-8590)\nA-16   
16                                                                     Proposition 28: Arts and Music In Schools (6770-8590)\nA-17   
17                                                                                        LCFF Equity Multiplier (7399-8590)\nA-18   
18                                                                   Student Support and Enrichment Block Grant (6018-8590) \nA-19   
19                                                                   New or Expanding Charter Advance Payments \n(0000-8011)\nA-20   
20  School District Advance Payments of In-lieu of Property Tax Transfers for New or Expanding Charter Schools \n(0000-8011)\nA-21   
21                                                                                                   Amount Charter Overpaid\nA-22   
22                                                                                                PY Amount Charter Overpaid\nA-23   
23                                                                                         Adjustments and PY Recomputations\nA-24   
24                                                                   Total Principal Apportionment\n(Sum of A-1 through A-24)\nB-1   
25                                                                                               EPA Entitlement\n(1400-8012)\nC-1   
26                                                       Total of LCFF State Aid and EPA Entitlement\n(A-1 + A-2 + A-3 + C-1)\nD-1   

    Amount in $  
0     740000620  
1   38246131573  
2    6319315906  
3       1394567  
4       6241697  
5             0  
6      50265280  
7       2983118  
8    4506685138  
9      90551361  
10    453169000  
11    259999998  
12   3995000000  
13            0  
14      8600000  
15    720630821  
16    907030402  
17    303210000  
18     31242360  
19   

In [171]:
summary_rows = ["County Office of Education LCFF State Aid \n(0000-8011)\nA-1",
                "School District LCFF State Aid\n (0000-8011)\nA-2",
                "Charter School LCFF State Aid\n (0000-8011)\nA-3",
                "Total Principal Apportionment\n(Sum of A-1 through A-24)\nB-1",
                "Special Education \nAB 602\n(6500-8311)\nA-9"]
summary = summary[summary['Funding Area'].isin(summary_rows)]
summary

Funding Area  Amount in $
0    County Office of Education LCFF State Aid \n(0000-8011)\nA-1    740000620
1               School District LCFF State Aid\n (0000-8011)\nA-2  38246131573
2                Charter School LCFF State Aid\n (0000-8011)\nA-3   6319315906
8                    Special Education \nAB 602\n(6500-8311)\nA-9   4506685138
24  Total Principal Apportionment\n(Sum of A-1 through A-24)\nB-1  56135473411

In [172]:
summary.to_csv("CA-Funding-Summary.csv")

done
